components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerAPIResourceManagementService.java

```diff
@@ -0,0 +1,475 @@
+                boolean isLastPage = !hasMoreItems && (StringUtils.isNotBlank(after) || StringUtils.isBlank(before));
+
+                String url = "?limit=" + limit;
+
+                if (StringUtils.isNotBlank(filter)) {
+                    try {
+                        url += "&filter=" + URLEncoder.encode(filter, StandardCharsets.UTF_8.name());
+                    } catch (UnsupportedEncodingException e) {
+                        LOG.error("Server encountered an error while building pagination URL for the response.", e);
+                    }
+                }
+
+                if (hasMoreItems) {
+                    apiResources.remove(apiResources.size() - 1);
+                }
+                if (needsReverse) {
+                    Collections.reverse(apiResources);
+                }
+                if (!isFirstPage) {
+                    String encodedString = Base64.getEncoder().encodeToString(apiResources.get(0).getCursorKey()
+                            .toString().getBytes(StandardCharsets.UTF_8));
+                    apiResourceListResponse.addLinksItem(buildPaginationLink(url + "&before=" + encodedString,
+                            "previous"));
+                }
+                if (!isLastPage) {
+                    String encodedString = Base64.getEncoder().encodeToString(apiResources.get(apiResources.size() - 1)
+                            .getCursorKey().toString().getBytes(StandardCharsets.UTF_8));
+                    apiResourceListResponse.addLinksItem(buildPaginationLink(url + "&after=" + encodedString, "next"));
+                }
+            }
+
+            if (apiResources == null || apiResources.isEmpty()) {
+                apiResourceListResponse.setTotalResults(0);
+                apiResourceListResponse.setApiResources(new ArrayList<>());
+                return apiResourceListResponse;
+            }
+            apiResourceListResponse.setTotalResults(apiResourceSearchResult.getTotalCount());
+            apiResourceListResponse.setApiResources(apiResourceSearchResult.getAPIResources().stream()
+                    .map(this::buildAPIResourceListItem).collect(Collectors.toList()));
+
+        } catch (APIResourceMgtException e) {
+            throw APIResourceMgtEndpointUtil.handleAPIResourceMgtException(e);
+        }
+        return apiResourceListResponse;
+    }
+
+    /**
+     * Get API Resource by ID.
+     *
+     * @param apiResourceID API Resource ID.
+     * @return API Resource.
+     */
+    public APIResource getAPIResourceById(String apiResourceID) {
+
+        try {
+            APIResource apiResource = APIResourceManagementServiceHolder.getApiResourceManager()
+                    .getAPIResourceById(apiResourceID, ContextLoader.getTenantDomainFromContext());
+            if (apiResource == null) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("API Resource not found for ID: " + apiResourceID);
+                }
```

Comment:
This debug log is unnecessary since the error throws all the details that are available in the debug logs. Either improve the log or remove the log

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/473/commits/5c51c9eff45d1c55a16c7858a056b5a9eed2ce86